### DISCUSSION TITLE
Default Location @ checkout

### DISCUSSION THREAD

**Discussion by in8dna** [openstore-ecommerce/OpenStore#139] [Q&A]
Is there a way to set the default dropdown region/state in the selection dropdown during checkout? It currently displays Alabama (alpha/numeric sort) and the client wants his state first in the list so customers do not have to make a selection. He only sells classes to residents of Texas.

CheckoutAddress.cshtml reads:
`

>                     <input id="ad_region" type="hidden" value="@info.GetXmlProperty("genxml/billaddress/genxml/dropdownlist/region")">

`
CheckoutRegion.cshtml reads:

> ```
> @{
>     var info = (NBrightInfo)Model.List.First();
>     var countrycode = info.GetXmlProperty("genxml/dropdownlist/country");
>     var xpath = info.GetXmlProperty("genxml/hidden/xpath");
> }
> 
> <label>@ResourceKey("General.lblRegion")</label>
> @RegionSelect(info, xpath, countrycode,"", false)
> ```

**Comment by leedavi** [comment]
I've never tried, but I think I would approach it on a change event and then set the default value you want using JS or Jquery.

**Comment by DNNMonster** [comment]
There is also a render address section in the checkout.js that should populate this data with your DNN user profile info.  If the profile address data doesn't get populated it will not work however.

You can try requiring your users to input their State(Region) in the DNN user profile settings when they register.  That way Open Store should populate it with the user's info.   

If you're not registering users ahead of time(and/or not requiring their Region input) this may not help and js or jQuery would likely be the better route.

**Reply by Rodien** [reply]
I think that you are looking for something like this => 



`<script>
$( document ).ready(function() {
	
	
	
	$('#billaddress_country option[value="NL"]').prop('selected', true);
    $('#shipaddress_country option[value="NL"]').prop('selected', true);

    $('#billaddress_region option[value="Country.NL.Region:ZH"]').prop('selected', true);
   

    $('#billaddress_country').attr("disabled", true); 
    $('#billaddress_region').attr("disabled", true); 
	$('#selectaddress > option').each(function() {
		var nt = $(this).text().replace(/,/g, " ")
		$(this).text(nt);
	});

});
</script>`




![image](https://user-images.githubusercontent.com/50739829/111391581-32896600-86b5-11eb-921a-2ea539b17c14.png)

**Comment by in8dna** [comment]
I cheated. I looked through various suggestions and thought of an 'old-school' way to solve the clients issue without code.

I added a row to the 'list' table with the 'state' preceded by a blank space. The default sort order is alpha-num so the blank space sorts first. That way. if they do not select a state, it's is the correct state (for the client) and if they select a state, it is also correct.

Simple, not very technically helpful.

I should try out Rodien's solution and see if that works as well.
